lib/chef/exceptions.rb

```diff
@@ -134,6 +134,13 @@ class InvalidVersionConstraint < ArgumentError; end
     # Version constraints are not allowed in chef-solo
     class IllegalVersionConstraint < NotImplementedError; end
 
+    class MetadataNotValid < StandardError; end
+    class MetadataNotFound < StandardError
+      def initialize
+        super "No metadata.rb or metadata.json!"
```

Comment:
I missed the previous patch that you're backporting, so maybe this should be a follow-on thing, but this error message isn't actionable enough IMO. Where is this cookbook? What is its name?